src/components/TutorialPopupComponent.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Component representing pop up for every reference and display dynamically the content of it
+ *
+ *
+ * @property {Object} components object that holds DOM elements that represent this component, as well as component's context
+ * @property {HTMLElement} components.questionMarkHighlight question mark for highlight feature tutorial
+ * @property {HTMLElement} components.questionMarkGraph question mark for graph feature tutorial
+ * @property {HTMLElement} components.questionMarkSummary question mark for summary feature tutorial
+ *  @property {HTMLElement} components.panelTutorial div that contains the tutorial window
+ */
+class TutorialPopupComponent {
```

Comment:
Why not separate the question marks in it's own class. That would be cleaner. And then for each question mark you have 1 object of that class.

Reply:
I honestly think that these question mark elements should not be here but in ToolbarComponent since their html is within the html of that component. I would then publish events inside mouseenter handlers event per each tutorial window and have a method in this component responsible for displaying just that one window, and have such method for each window. This is the example for _highlight tutorial window_ but can be applied for all other tutorial windows
In #registerEvents method of ToolbarComponent i would add
   this.components.questionMarkHighlight.addEventListener(
      "mouseenter",
      (event) => {
        event.preventDefault();
        EventHandlerService.publish(PDFLEvents.onShowHighlightTutorital) //myb think of better name for this event
      }
    );
    this.components.questionMarkHighlight.addEventListener(
      "mouseleave",
      (event) => {
        event.preventDefault();
        EventHandlerService.publish(PDFLEvents.onHideTutorialWindow) 
      }
    );

and then in #registerEvents method TutorialPopupComponent i would have:
EventHandlerService.subscribe(
      PDFLEvents.onShowHighlightTutorital,
      this.#showHighlightTutorial();
    );
EventHandlerService.subscribe(
      PDFLEvents.onHideTutorialWindow,
      this.#hideTutorialWindow();
    );

Then i would take advantage of "tutorial-window" class which highlight-tutorial-window, graph-tutorial-window and summary-tutorial-window all have and create a method in TutorialPopupComponent called something like cleanWindow which would look something like like:
cleanWindow(){
   document.querySelectorAll(".tutorial-window").forEach(e=>e.classList.add("hidden"));
}
This method can then also be called in #hideTutorialWindow method instead of explicitly hiding elements by id.

Then, in showHighlightTutorial i would call that method and remove hidden just from highlight-tutorial-window element, so something like :
showHighlightTutorial(){
   this.cleanWindow();
   this.components.highlightTutorialWindow.classList.remove("hidden");
}

This way, adding a new tutorial window is easy because you do not have to change logic of existing code, but just add to it and you would also lose the ugly switch case.

Reply:
Also, you're sometimes referring to this component in both documentation and code as a popup, sometimes as a window and sometimes as panel - choose one and your documentation and code will make far more sense.